File: Awakened_talents_admin/src/app/models/TeacherProfile.ts
import { Subject } from "../models/Subjects";
import { User } from "./User";

export class TeacherProfile {
    Id: string;
	Description: string;
    User: User;
    Subjects: Subject[];
    constructor() {
        this.Id = ""
        this.Description = ""
        this.User = null!;
        this.Subjects = [];
    }
}

File: AwakenedTalents/AwakenedTalents/Models/TeacherProfile.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;

namespace AwakenedTalents.Models
{
	public class TeacherProfile
	{
		public TeacherProfile() {
			Subjects = new List<Subject>();
		}

        [Key]
        [ForeignKey("User")]
        public string Id { get; set; }
		public User User { get; set; }
		public virtual ICollection<Subject> Subjects { get; set; }
		public string Description { get; set; }
	}
}
